### PR TITLE
Trailing whitespace detection script and circleci job.

### PR DESCRIPTION
**Pull request by ekpyron**

### CODE REVIEW

File: scripts/detect_trailing_whitespace.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
```
Comment (chriseth) [inline scripts/detect_trailing_whitespace.sh]: Are you sure sh is enough?

File: scripts/detect_trailing_whitespace.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+set -e
+
+REPO_ROOT="$(dirname "$0")"/..
+
+cd $REPO_ROOT
```
Comment (chriseth) [inline scripts/detect_trailing_whitespace.sh]: Please put this into `(` so that the wd of the calling process is not altered.

**Comment by chriseth** [comment]
`/usr/bin/env: 'bash -e': No such file or directory`

**Comment by ekpyron** [comment]
It says "[WIP]" for a reason ;-).

**Comment by chriseth** [comment]
Ah, didn't see that ;)

**Comment by ekpyron** [comment]
Now it should be fine and successfully detected the whitespace in ``ExpressionCompiler.cpp`` before the second commit and now correctly passes.

**Comment by ekpyron** [comment]
We could actually think about doing the same for leading spaces instead of tabs... that would probably be more involved, though, since there are places, in which that is valid.

**Review by chriseth — APPROVED** [review]

**Pull request merged**